File: Curs01/Python/02-sum.py
#!/usr/bin/env python3
"""
   Programarea Calculatoarelor CA, 2019-2020

   Programul calculeaza suma a 2 numere și afisează rezultatul în baza 10 și în baza 16.
"""


def main():
    # Citim primul rand din input (un șir de caractere), împărțim rezultatul după spațiu,
    # apoi transformăm fiecare componentă în număr întreg folosind map.
    # Rezultatul (2 int-uri, dacă input-ul este corect) se asignează variabilelor a și b.
    a, b = map(int, input().split())

    # Calculăm suma dintre a și b
    sum = a + b

    # Afișăm rezultatul în baza 10, cu un mesaj corespunzător
    print('Baza zece: {}'.format(sum))
    # Puteam folosi și print('Baza zece: ' + str(sum)), dar varianta cu format este mai flexibilă

    # Afișăm rezultatul în baza 16 (folosind litere mici)
    print('Baza 16: {:x}'.format(sum))

    # Afișăm rezultatul în baza 16 (folosind litere mari)
    print('Baza 16: {:X}'.format(sum))

if __name__ == '__main__':
    main()

# rulare: ./02-sum.py sau python3 ./02-sum.py
# Pentru prima variantă, este necesar să avem drepturi de execuție asupra fișierului.
#
# Exemplu:
#
# input:
# 3 8
# output:
# Baza zece: 11
# Baza 16: b
# Baza 16: B

File: Curs01/C/02-sum.c
/* Programarea Calculatoarelor CA, 2019-2020

   Programul calculeaza suma a 2 numere si afiseaza rezultatul 
   in baza 10 si in baza 16 */

#include <stdio.h>

int main() {
    int a, b, sum;
    scanf("%d%d", &a, &b);

    sum = a + b;

    // %d se foloseste pentru afisarea unui intreg in baza 10
    printf("Baza 10: %d\n", sum);
    // %X se foloseste pentru afisarea unui intreg in baza 16 (litere mari)
    printf("Baza 16: %X\n", sum);
    // %x se foloseste pentru afisarea unui intreg in baza 16 (litere mici)
    printf("Baza 16: %x\n", sum);
    return 0;
}

/* compile: gcc 02-sum.c -o sum
   run    : ./sum

   Exemplu:

   input:
   3 8
   output:
   Baza 10: 11
   Baza 16: B  
   Baza 16: b  */
